UnicornEngine/video/source/geometry/Cube.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+* Copyright (C) 2017 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+
+#include <unicorn/video/geometry/Cube.hpp>
+#include <unicorn/video/Color.hpp>
+
+namespace unicorn
+{
+namespace video
+{
+namespace geometry
+{
+Cube::Cube(std::shared_ptr<Mesh> mesh)
+    : MeshDescriptor(mesh)
+{
+    std::vector<Vertex> temp_vertices;
+
+    //front
+    temp_vertices.push_back({{-1.0, -1.0, 1.0}, {Color::White}});
```

Comment:
If push_back take rvalue, it simply move it inside vector. Absolutely similar to emplace back.
Difference only if we pass lvalue